draft-ietf-tls-tls13.md

```diff
@@ -3488,6 +3503,9 @@ access_denied
 decode_error
 : A message could not be decoded because some field was out of the
   specified range or the length of the message was incorrect.
+  This alert is used for errors where the message does not conform
+  to the formal protocol syntax (e.g., a vector is specified as
+  having the range <2..255> but is encoded as having 0 length.)
```

Comment:
I think it's confusing to try to think about "only received 3 fields when 4 fields expected"; it's more like "only received enough data for 3 fields of the 4-field struct expected, which may indicate the sender sent a different type than was expected", since the bits on the wire are not tagged with an identifier for what structure they represent.